packages/appium/lib/cli/setup-command.js

```diff
@@ -0,0 +1,233 @@
+ * @param {string} appiumHome
+ * @param {DriverConfig} driverConfig
+ * @param {PluginConfig} pluginConfig
+ * @returns {Promise<void>}
+ */
+export async function runSetupCommand(appiumHome, preConfigArgs, driverConfig, pluginConfig) {
+  switch (preConfigArgs.setupCommand) {
+    case SUBCOMMAND_RESET:
+      await resetAppiumHome(appiumHome, driverConfig, pluginConfig);
+      break;
+    case SUBCOMMAND_DESKTOP:
+      await setupDesktopAppDrivers(driverConfig);
+      await setupDefaultPlugins(pluginConfig);
+      break;
+    case SUBCOMMAND_BROWSER:
+      await setupBrowserDrivers(driverConfig);
+      await setupDefaultPlugins(pluginConfig);
+      break;
+    default:
+      await setupMobileDrivers(driverConfig);
+      await setupDefaultPlugins(pluginConfig);
+      break;
+  }
+};
+
+/**
+ * Uninstall installed drivers and plugins in APPIUM_HOME. Then remove the entire Appium home.
+ * @param {string} appiumHome
+ * @param {DriverConfig} driverConfig
+ * @param {PluginConfig} pluginConfig
+ * @returns {Promise<void>}
+ */
+async function resetAppiumHome(appiumHome, driverConfig, pluginConfig) {
+  log.info(`Uninstalling drivers/plugins in APPIUM_HOME[${appiumHome}]`);
+  await uninstallDrivers(driverConfig);
+  await uninstallPlugin(pluginConfig);
+  await fs.rimraf(appiumHome);
+}
+
+/**
+ * Install drivers listed in DEFAULT_DRIVERS.
+ * @param {DriverConfig} driverConfig
+ * @returns {Promise<void>}
+ */
+async function setupMobileDrivers(driverConfig) {
+  await installDrivers(SUBCOMMAND_MOBILE, driverConfig);
+}
+
+/**
+ * Install all of known drivers listed in BROWSER_DRIVERS.
+ * @param {DriverConfig} driverConfig
+ * @returns {Promise<void>}
+ */
+async function setupBrowserDrivers(driverConfig) {
+  await installDrivers(SUBCOMMAND_BROWSER, driverConfig);
+}
+
+/**
+ * Install all of known drivers listed in DESKTOP_APP_DRIVERS.
+ * @param {DriverConfig} driverConfig
+ * @returns {Promise<void>}
+ */
+async function setupDesktopAppDrivers(driverConfig) {
+  await installDrivers(SUBCOMMAND_DESKTOP, driverConfig);
+}
+
+/**
+ * Install the given driver name. It skips the installation if the given driver name was already installed.
+ * @param {import('appium/types').CliCommandSetupSubcommand} subcommand
+ * @param {DriverConfig} driverConfig
+ * @returns {Promise<void>}
+ */
+async function installDrivers(subcommand, driverConfig) {
+  for (const driverName of getPresetDrivers(subcommand)) {
+    await installExtention(driverName, extensionCommandArgs('driver', driverName, 'install'), driverConfig);
+  }
+}
+
+/**
+ * Uninstall the given driver name.
+ * @param {DriverConfig} driverConfig
+ * @returns {Promise<void>}
+ */
+async function uninstallDrivers(driverConfig) {
+  for (const driverName of _.keys(driverConfig.installedExtensions)) {
+    try {
+      await runExtensionCommand(extensionCommandArgs('driver', driverName, 'uninstall'), driverConfig);
+    } catch (ign) {}
```

Comment:
maybe it makes sense to still log the error and the stacktrace

Reply:
same below

Reply:
Uninstallation-related stuff will be in https://github.com/appium/appium/issues/20117